src/formatDate.js

```diff
@@ -50,7 +50,40 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[fromFormat.length - 1]);
+  const newDate = [0, 0, 0];
+  const dateInfo = {};
+  let count = 0;
+
+  let year = 0;
+
+  for (const key of fromFormat) {
```

Comment:
@Patrykmclaren14 please apply previous reviewers' suggestions here, and your task will be approved ;)